### PR TITLE
 [PLUGIN-1105] Updated Mask transformation

### PR DESCRIPTION
**Pull request by itsankit-google**
[PLUGIN-1105](https://cdap.atlassian.net/browse/PLUGIN-1105)

### CODE REVIEW

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java [hunk 1]
```diff
@@ -49,30 +66,32 @@ private void parseDirective(TransformationContext context) {
...
+                                           "direction mask_char n.");
     }
 
     srcColumn = splits[1];
-    try {
-      fromPos = Integer.parseInt(splits[2]);
-    } catch (NumberFormatException e) {
-      throw new IllegalArgumentException("from_position is not an integer.", e);
+    direction = splits[2];
+    if (!(direction.equalsIgnoreCase(RIGHT_DIRECTION) || direction.equalsIgnoreCase(LEFT_DIRECTION))) {
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/transformation/MaskTransformation.java]: if we are making directive name lower case, then we should also make direction lower case to be consistent.

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java [hunk 2]
```diff
@@ -49,30 +66,32 @@ private void parseDirective(TransformationContext context) {
...
+      throw new IllegalArgumentException(String.format("masking direction should be %s or %s",
+                                                       RIGHT_DIRECTION, LEFT_DIRECTION));
+    }
+    
+    maskCharacter = splits[3];
+    if (maskCharacter.length() != 1) {
```
Comment (seanzhougoogle) [inline src/main/java/io/cdap/delta/transformation/MaskTransformation.java:87]: have we decided what are the characters limitation ? is Unicode allowed , unicode may not be 1 character length.

**Review by seanzhougoogle — APPROVED** [review]
LGTM with minor comments.

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java (hunk 1 above)
Comment (sahusanket) [reply src/main/java/io/cdap/delta/transformation/MaskTransformation.java]: +1

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java (hunk 2 above)
Comment (itsankit-google) [reply src/main/java/io/cdap/delta/transformation/MaskTransformation.java:87]: @bdmogal WDYT?

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java (hunk 1 above)
Comment (itsankit-google) [reply src/main/java/io/cdap/delta/transformation/MaskTransformation.java]: ack.

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java (hunk 2 above)
Comment (bdmogal) [reply src/main/java/io/cdap/delta/transformation/MaskTransformation.java:87]: I think its fine to not allow unicode for now. We can relax this restriction to support unicode if we get a use case for it in future.

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java [hunk 3]
```diff
@@ -49,30 +66,32 @@ private void parseDirective(TransformationContext context) {
...
+      throw new IllegalArgumentException(String.format("masking_character: %s is not a character", maskCharacter));
     }
 
     try {
-      toPos = Integer.parseInt(splits[3]);
+      countN = Integer.parseInt(splits[4]);
```
Comment (sahusanket) [inline src/main/java/io/cdap/delta/transformation/MaskTransformation.java:92]: Should be whole number ( >=0 ) 

I think it should NOT be negative. right?

Comment (itsankit-google) [reply src/main/java/io/cdap/delta/transformation/MaskTransformation.java:92]: Good catch, added a check and unit test for this case.

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java [hunk 4]
```diff
@@ -49,30 +66,35 @@ private void parseDirective(TransformationContext context) {
...
+      if (!(countN >= 0)) {
+        throw new IllegalArgumentException(String.format("n is not a whole number, given: %s", countN));
```
Comment (sahusanket) [inline src/main/java/io/cdap/delta/transformation/MaskTransformation.java:94]: readability nit :  `(countN < 0)`

**Review by sahusanket — APPROVED** [review]

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java (hunk 4 above)
Comment (itsankit-google) [reply src/main/java/io/cdap/delta/transformation/MaskTransformation.java:94]: ack

**Pull request merged**